ugs-platform/ProbeModule/src/main/java/com/willwinder/ugs/platform/probe/ProbeService.java

```diff
@@ -378,6 +381,91 @@ private void performXYZProbeInternal(int stepNumber) throws IllegalStateExceptio
         }
     }
 
+    void performHoleCenterProbe(ProbeParameters params) throws IllegalStateException {
+        validateState();
+        currentOperation = ProbeOperation.INSIDE_CIRCLE;
+        this.params = params;
+        performHoleCenterProbeInternal(0);
+    }
+
+    private void performHoleCenterProbeInternal(int stepNumber) throws IllegalStateException {
+        String g = GcodeUtils.unitCommand(params.units);
+        String g0Abs = "G90 " + g + " G0";
+        String g0Rel = "G91 " + g + " G0";
+        String g0MCS = "G53 " + g + " G0";
+        double holeRadius = params.holeDiameter / 2.0;
+
+        continuation = () -> performHoleCenterProbeInternal(stepNumber + 1);
+
+        try {
+            switch (stepNumber) { // NOTE: G code comments are with radius 25 and retract 2, G21, G54
+                case 0: { // find -X
+                    // Reset (0,0,_) to make it easier to retract.
+                    updateWCS(params.wcsToUpdate, 0.0, 0.0, null); // G10 L20 P0 X0 Y0
+                    probe('X', params.feedRate, -holeRadius, params.units); // G21 G91 G49; G38.2 X-25.0 F250
+                    break;
+                }
+                case 1: { // retract & measure -X
+                    gcode(g0Rel + " X" + params.retractAmount); // G91 G21 G0 X2.0
```

Comment:
I can see that we previously did it this way, just concatinating strings and doubles.

The problem is that this is locale sensitive. In some locales a minus character is UTF-8 encoded using two bytes. Commas may also be represented with a `,` instead of a `.`.

Can you replace your concatinations of doubles using `Utils.formatter.format(x)` or similar instead now that you have code fresh?

Reply:
Thanks for review, forgot the locales stuff... Should be fixed now.